src/reconcile/rec_write.c

```diff
@@ -187,35 +193,35 @@ typedef struct {
 		 * column-store key.
 		 */
 		WT_ITEM key;		/* Promoted row-store key */
+
+		/*
+		 * For each boundary that we create at the split percentage, we
+		 * also maintain details of where a boundary would be, if we
+		 * split at the minimum split percentage.
+		 * While reconciling pages, at any given time, we maintain two
+		 * split chunks in the memory to be written out as pages. As we
+		 * get to the last two chunks, if the last one turns out to be
+		 * smaller than the minimum split size, we go back into the
+		 * penultimate chunk and split at this minimum split size
+		 * boundary. This moves some data from the penultimate chunk to
+		 * the last chunk, hence increasing the size of the last page
+		 * written without decreasing the penultimate page size beyond
+		 * the minimum split size.
+		 */
+		uint32_t min_bnd_entries; /* No of entries at min split pct */
+		/* Next chunk's starting offset at min split pct */
+		size_t min_bnd_offset;
+		/* Next chunk's starting record number at min split pct */
+		uint64_t min_bnd_recno;
+		/* Next chunk's starting key at min split pct */
+		WT_ITEM min_bnd_key;
```

Comment:
renamed to max/min. Had a discussion with Keith, if we want one set of min_XXX, will need to add an additional field that will say if for current chunk minimum boundary has been set or not. We decided to not make that change for now. Will follow up in another ticket on improvements that we can make on top of this change.